packages/schematics/angular/application/files/src/main.ts

```diff
@@ -1,12 +1,25 @@
-import { enableProdMode } from '@angular/core';
-import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
+<%= experimentalIvy ? '// ' : '' %>import { enableProdMode } from '@angular/core';
+<%= experimentalIvy ? '// ' : '' %>import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
+<%= experimentalIvy ? '// ' : '' %>
+<%= experimentalIvy ? '// ' : '' %>import { AppModule } from './app/app.module';
+<%= experimentalIvy ? '// ' : '' %>import { environment } from './environments/environment';
+<%= experimentalIvy ? '// ' : '' %>
+<%= experimentalIvy ? '// ' : '' %>if (environment.production) {
+<%= experimentalIvy ? '// ' : '' %>  enableProdMode();
+<%= experimentalIvy ? '// ' : '' %>}
+<%= experimentalIvy ? '// ' : '' %>
+<%= experimentalIvy ? '// ' : '' %>platformBrowserDynamic().bootstrapModule(AppModule)
+<%= experimentalIvy ? '// ' : '' %>  .catch(err => console.log(err));
+<% if (experimentalIvy) { %>
+import { AppComponent } from './app/app.component';
+import { ɵrenderComponent as renderComponent } from '@angular/core';
 
-import { AppModule } from './app/app.module';
 import { environment } from './environments/environment';
 
 if (environment.production) {
-  enableProdMode();
+  // Force typing on the window object to remove linting or typing errors.
+  (window as {} as { ngDevMode: boolean }).ngDevMode = false;
```

Comment:
we can omit this now. right?

even without the fix from Misko, we'll be able to avoid this if we set the ngDevMode as compile time constant when the `experimentalIvy` flag is on

Reply:
We'll have to check the `enableIvy` flag instead, but yes. Removed.